### DISCUSSION TITLE
terragrunt local and remote backend

### DISCUSSION THREAD

**Discussion by kkumaar999** [gruntwork-io/knowledge-base#505] [Help] [answered]
We have one terragrunt backend configured in gcs bucket. We are using the same bucket for all the executions. Can we make this remote state as optional i.e when user wants the state to be created local instead of already created backend.

---

<ins datetime="2022-07-15T07:22:05Z">
  <p><a href="https://support.gruntwork.io/hc/requests/108988">Tracked in ticket #108988</a></p>
</ins>

**Comment by yorinasub17** [comment, marked as answer]
Unfortunately, there really isn't a great way to implement this in terragrunt right now because there isn't a way to turn off a block. The best way I can think of is to have users/devs uncomment a `remote_state` block in the child that overrides the remote state block in the parent.

There might be a better way using conditionals and exposed includes, but then there will be a good amount of copy paste.
